src/protocols/secure_channel/PASESession.cpp

```diff
@@ -213,17 +213,18 @@ CHIP_ERROR PASESession::Pair(SessionManager & sessionManager, uint32_t peerSetUp
 {
     MATTER_TRACE_SCOPE("Pair", "PASESession");
     ReturnErrorCodeIf(exchangeCtxt == nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    ExchangeHandle ecHandle(*exchangeCtxt);
```

Comment:
```suggestion
```
We don't use ecHandle anywhere other than the SetValue() call; why do we need it here?  And if you use a temporary ExchangeHandle, you don't have to worry about the goto bits here.